### ISSUE TITLE
WooCommerce navigation: it's not possible to open react pages in new tabs

### ISSUE THREAD

**Issue by Aljullu** [woothemes/woocommerce#52120]
It's not possible to open WooCommerce Analytics pages into new tabs via <kbd>Ctrl</kbd> + Click.

## Steps to reproduce

1. Go to _Analytics_.
2. <kbd>Ctrl</kbd> + Click on _Products_.
3. Notice the page opens on the current screen, instead of creating a new tab.

https://github.com/user-attachments/assets/4ca8d21e-b3f3-4206-9c24-3584cb17b00f

_(👆 while it's not visible in the video, I have <kbd>Ctrl</kbd> pressed while clicking)_

## Expected behavior

Clicking on a link with <kbd>Ctrl</kbd> pressed should open the link in a new tab.

**Comment by tomalec** [comment]
I +1 this issue, as I personally find the lack of <kbd>Ctrl</kbd> + `click` disturbing as well.

However, this happens for all react-based menu entries, like 
 - WooCommerce > Home
 - WooCommerce > Customers
 - WooCommerce > Extensions

So, in my opinion, it's more of a core navigation issue than an analytics issue.

**Comment by Manussakis** [comment]
I have started working on this issue.

**Comment by Manussakis** [comment]
I am having a hard time debugging this issue.

According to my assessment, there are two fixes needed:

1. Open the page in a new tab when `Ctrl` or `Command` (on Mac) is pressed along with the click.
2. Prevent the page from opening on the current tab.

If only item 1 were required, the fix should be easy. Adding the script below to the [client/admin/client/index.js](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/client/admin/client/index.js) file does the trick:

```js
( function () {
	const adminmenu = document.getElementById( 'adminmenu' );

	if ( ! adminmenu ) {
		return;
	}

	const wcMenuItemListener = ( event ) => {
		const closestWCAdminMenuItem = event.target.closest(
			'a[href*="page=wc-admin"]'
		);

		if ( ! closestWCAdminMenuItem ) {
			return;
		}

		if ( event.ctrlKey || event.metaKey ) {
			event.preventDefault();
			window.open( closestWCAdminMenuItem.href, '_blank' );
		}
	};

	adminmenu.addEventListener( 'click', wcMenuItemListener );
} )();
```

However, preventing the clicked page from opening on the current browser tab is more complicated. According to my investigation, this functionality is managed by react-router-dom, so it would be necessary to be in a react-based environment to intercept the click event and decide whether it should open on the current page.

This requires some knowledge of adding new WooCommerce admin pages, which I don't have much experience with as a frontend engineer.

I would appreciate any guidance on this issue.

Thanks!

cc: @vikrampm1

**Comment by vikrampm1** [comment]
@Aljullu @tomalec any thoughts or guidance on the above?

**Comment by Aljullu** [comment]
> According to my assessment, there are two fixes needed:
> 
> 1. Open the page in a new tab when `Ctrl` or `Command` (on Mac) is pressed along with the click.
> 2. Prevent the page from opening on the current tab.

To me, 1 and 2 are required; otherwise the behavior might feel a bit confusing. So, in the end, it should behave like any other link.

Unfortunately, I'm not very familiar with that part of the codebase, I took a brief look, but I couldn't find out easily where this navigation logic is handled. @rjchow I see you are Ghidorah's porter this week, wondering if you have some insights that can help?

**Comment by rjchow** [comment]
cc: @woocommerce/ghidorah

**Comment by ilyasfoo** [comment]
Hi @vikrampm1. I believe the code responsible for applying react navigation can be [found here](https://github.com/woocommerce/woocommerce/blob/3187e128de683c91c907b04092700fdbb1448d94/plugins/woocommerce/client/admin/client/layout/controller.js#L464-L487). I think adding the checks prior `getHistory().push( href )` should prevent current tab from navigating away

**Comment by Manussakis** [comment]
Thanks, @ilyasfoo 🙏
This helps a lot!
